crates/sui-indexer-alt-graphql/src/api/types/object.rs

```diff
@@ -234,33 +244,83 @@ impl Object {
         address: SuiAddress,
         version: UInt53,
     ) -> Result<Option<Self>, RpcError<Error>> {
-        let contents = ObjectContents::fetch(ctx, address, version).await?;
-        let Some(object) = &contents.0 else {
+        let Some(object) = contents(ctx, address, version).await? else {
             return Ok(None);
         };
 
         Ok(Some(Object {
             address: object.id().into(),
             version: object.version(),
             digest: object.digest(),
-            contents,
+            contents: Some(object),
         }))
     }
+
+    /// Return a copy of the object's contents, either cached in the object or fetched from the KV
+    /// store.
+    async fn contents(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Option<Arc<NativeObject>>, RpcError<Error>> {
+        if self.contents.is_some() {
+            Ok(self.contents.clone())
+        } else {
+            contents(ctx, self.address.into(), self.version.into()).await
```

Comment:
I see what you meant in the commit message now. This code path is hit every time this object's contents are requested since we never fill `self.contents`. Luckily there's data loader for caching.